### PR TITLE
Call get-room-permission edge function to get user's role on a room

### LINKED ISSUE #92: FE: Call get-room-permission edge function when entering a room
Call get-room-permission when a user enters a room in order to get its role.

### PR DESCRIPTION
**Pull request by clararfe**
**WIP:** waiting for storage to be ready to consume necessary data from there.

The edge function get-room-permission gives the role a user has on a specific room. 
This PR contains the implementation of calling the above mention edge function from the frontend everytime a user joins a room.

**TODO:**
If the storage is also gonna save the role of the user on each room, this code can be improved: call the edge function only the first time the user joins the room, otherwise get the role from storage.

### CODE REVIEW

**Comment by marcovidonis** [comment]
I think that we might still need to call the function every time a user enters a room, unless we get from storage that they're an admin. That's because everyone else could have a different role in each room, and also their role could be changed at any point by someone with a higher role.

File: frontend/src/actions/index.js [hunk 1]
```diff
@@ -139,3 +139,37 @@ export const roomJWTprovider = async (
...
+      },
+    );
+    onSuccess && onSuccess(response);
+    /* eslint-disable consistent-return */
+    return response.data;
+  } catch (error) {
+    if (error.response.status === 404) {
+      onNotFound && onNotFound();
+    } else if (error.response.status !== 200) {
+      throw new Error(`unexpected ${error.response.status} response`);
```
Comment (marcovidonis) [inline frontend/src/actions/index.js]: Should we call `onError()` here, too, like the other actions?

**Comment by clararfe** [comment]
> I think that we might still need to call the function every time a user enters a room, unless we get from storage that they're an admin. That's because everyone else could have a different role in each room, and also their role could be changed at any point by someone with a higher role.

Yeah, sounds the good way to call the edge function every time a user enters a room.

**Comment by clararfe** [comment]
Final approach: the edge function is called every time the user joins a room so no data needs to be stored.

**Pull request closed without merging**